guidelines/UnitTests

```diff
@@ -0,0 +1,69 @@
+How to add unit test
```

Comment:
Is this even still relevant? `tests/unit/` has 3 commits, all from december 2016, and contains no actual tests.

Reply:
The associated code still exists so it's there for now - it's by no means the only thing horribly out of date in these files.  (The unit test framework was a drive-by PR from someone who promised to go on to add many unit tests for yosys but then never did. It should probably be removed altogether.)